app/api/microservice.py

```diff
@@ -216,15 +219,61 @@ def delete_microservice(
             status_code=404, detail="Can't delete Microservice ScoreCard")
 
     return {"message": "Microservice and associated scorecards successfully deleted"}
- 
- 
+
+
 @router.get("/{service_id}/metric_reading", response_model=list[ServiceMetricReading])
-def get_metrics(service_id: int, from_date: Optional[datetime] = None, 
-    to_date: Optional[datetime] = None,  service_metric_crud: CRUDServiceMetric = Depends(dependencies.getServiceMetricsCrud)):
+def get_metrics(service_id: int, from_date: Optional[datetime] = None,
+                to_date: Optional[datetime] = None,  service_metric_crud: CRUDServiceMetric = Depends(dependencies.getServiceMetricsCrud)):
+
+    metrics = service_metric_crud.get_metric_values_by_service(
+        service_id, from_date, to_date)
 
-    metrics = service_metric_crud.get_metric_values_by_service(service_id, from_date,to_date)
-    
     if not metrics:
-        raise HTTPResponseCustomized(status_code=404, detail="Metrics not found for this service and metric.")
+        raise HTTPResponseCustomized(
+            status_code=404, detail="Metrics not found for this service and metric.")
+
+    return metrics
+
+
+@router.post("/{service_id}/{metric_id}/reading", response_model=None)
+def create_metric_reading(
+    service_id: int,
+    metric_id: int,
+    newmservicemetric: ServiceMetricCreate,
+    microservice: CRUDMicroservice = Depends(
+        dependencies.getMicroservicesCrud),
+    servicemetric: CRUDServiceMetric = Depends(
+        dependencies.getServiceMetricsCrud),
+    metric: CRUDMetric = Depends(dependencies.getMetricsCrud),
+):
+
+    microservice_obj = microservice.get(service_id)
+    if not microservice_obj:
+        raise HTTPResponseCustomized(
+            status_code=404, detail="Service not found")
+
+    metric_obj = metric.get(metric_id)
+    if not metric_obj:
+        raise HTTPResponseCustomized(
+            status_code=404, detail="Metric not found")
+
+
+    metric_value = utity_datatype.parse_stringified_value(
```

Comment:
parse_stringified_value raises a Value error if the value doesn't align with the metric type.
shouldn't you add a try except block and in case of ValueError exception raise our custome http exception with descriptive error message ?

Reply:
try:
        metric_value = utity_datatype.parse_stringified_value(newmservicemetric.value, metric_obj.type)
    except ValueError as e:
        raise HTTPException(
            status_code=400,
            detail=f"Invalid value for metric '{metric_obj.name}': {str(e)}. Expected type: {metric_obj.type}"
        )

Reply:
are you need this ?